src/LRS.js

```diff
@@ -887,6 +888,42 @@ TinCan client library
         },
 
         /**
+        Assigns attachment content to the correct attachment to create a StatementsResult object that is sent
+        to the callback of queryStatements()
+
+        @method _assignAttachmentContent
+        @private
+        @param {Array} [stmts] Array of TinCan Statement JSON objects
+        @param {Array} [headers] Array containing section headers from the HTTP response
+        @param {String} [content] The content to place into its attachment
+        @return {Array} Array of TinCan Statement JSON objects with correctly assigned attachment content
+        */
+        _assignAttachmentContent: function (stmts, headers, content) {
```

Comment:
I wonder if it makes more sense to collect all attachments based on hash, create a map of them, then associate with the statements via the map which should be faster because you'll only run through the statements + attachment lists once, with the current implementation you have to loop over the full list of attachments for each attachment. Granted the lists are small, but no reason to include the overhead AFAICT.